### PR TITLE
feat: create global package

### PR DESCRIPTION
**Pull request by himself65**
Upstream: https://github.com/toeverything/blocksuite/pull/617

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "@commitlint/cli": "^17.4.0",
     "@commitlint/config-conventional": "^17.4.0",
     "@playwright/test": "^1.29.1",
+    "@types/blocksuite__global-types": "workspace:*",
```
Comment (doodlewind) [inline package.json:38]: Is there any documentation about this `__` syntax, or is it just a naming convention?

Comment (himself65) [reply package.json:38]: No, just study from @Jack-Works!

Comment (himself65) [reply package.json:38]: Refs: https://github.com/DimensionDev/Maskbook/blob/fdd4a551d468244ce0a7252c47efc7f5639c8b49/tsconfig.json#L29

**Comment by himself65** [comment]
Blocking as I need time to remove all `shape block` related code

**Comment by doodlewind** [comment]
Vercel error log:

```
> @blocksuite/next-example@ build /vercel/path0/packages/react/examples/next
--
13:56:25.675 | > next build
13:56:25.675 |  
13:56:26.153 | info  - Linting and checking validity of types...
13:56:26.581 | warn  - TypeScript project references are not fully supported. Attempting to build in incremental mode.
13:56:33.505 | info  - Creating an optimized production build...
13:56:38.263 | Failed to compile.
13:56:38.264 |  
13:56:38.264 | ./components/WorkspacesDropdown.tsx
13:56:38.264 | Module not found: Can't resolve '@blocksuite/blocks/models' in '/vercel/path0/packages/react/examples/next/components'
13:56:38.264 |  
13:56:38.264 | Import trace for requested module:
13:56:38.264 | ./components/WorkspacesDropdown.tsx
13:56:38.265 |  
13:56:38.265 | ./pages/_app.tsx
13:56:38.265 | Module not found: Can't resolve '@blocksuite/blocks/models' in '/vercel/path0/packages/react/examples/next/pages'
13:56:38.265 |  
13:56:38.265 | ../../src/store/index.ts
13:56:38.265 | Module not found: Can't resolve '@blocksuite/blocks/models' in '/vercel/path0/packages/react/src/store'
13:56:38.265 |  
13:56:38.265 | Import trace for requested module:
13:56:38.265 | ../../src/store/index.ts
13:56:38.266 | ../../src/index.ts
13:56:38.266 |  
13:56:38.266 | ../../src/store/manager/index.ts
13:56:38.266 | Module not found: Can't resolve '@blocksuite/blocks/models' in '/vercel/path0/packages/react/src/store/manager'
13:56:38.266 |  
13:56:38.266 | Import trace for requested module:
13:56:38.266 | ../../src/store/manager/index.ts
13:56:38.266 | ../../src/store/index.ts
13:56:38.266 | ../../src/index.ts
13:56:38.266 |  
13:56:38.266 |  
13:56:38.266 | > Build failed because of webpack errors
13:56:38.285 | ELIFECYCLE  Command failed with exit code 1.
13:56:38.300 | Error: Command "pnpm run build" exited with 1


```

File: package.json (hunk 1 above)
Comment (Jack-Works) [reply package.json:38]: It's documented in https://github.com/DefinitelyTyped/DefinitelyTyped

**Review by doodlewind — APPROVED** [review]

File: packages/react/examples/next/next.config.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const path = require('node:path');
+const baseDir = path.resolve(__dirname, '..', '..', '..', '..');
```
Comment (lawvs) [inline packages/react/examples/next/next.config.js:3]: Do you want `process.cwd()`?

**Pull request merged**